File: app/src/main/java/by/offvanhooijdonk/tofreedom/helper/surprise/IEvent.java
package by.offvanhooijdonk.tofreedom.helper.surprise;

import android.app.Activity;
import android.content.Context;
import android.support.v7.app.ActionBar;
import android.view.ViewGroup;

public interface IEvent {
    void run();

    interface IEventBuilder {
        IEventBuilder context(Context context);

        IEventBuilder rootView(ViewGroup root);

        IEventBuilder activity(Activity activity);

        IEventBuilder actionBar(ActionBar actionBar);

        IEvent build();
    }
}

File: app/src/main/java/by/offvanhooijdonk/tofreedom/helper/surprise/IEvent.kt
package by.offvanhooijdonk.tofreedom.helper.surprise

import android.app.Activity
import android.content.Context
import android.support.v7.app.ActionBar
import android.view.ViewGroup

interface IEvent {
    fun run()

    interface IEventBuilder {
        fun context(context: Context): IEventBuilder

        fun rootView(root: ViewGroup): IEventBuilder

        fun activity(activity: Activity): IEventBuilder

        fun actionBar(actionBar: ActionBar): IEventBuilder

        fun build(): IEvent
    }
}
